sopel/modules/safety.py

```diff
@@ -85,13 +122,18 @@ def setup(bot):
             # Python on Windows throws an exception if the file is in use
             LOGGER.info('Could not delete %s: %s', old_file, str(err))
 
-    loc = os.path.join(bot.config.homedir, 'unsafedomains.txt')
-    if os.path.isfile(loc):
-        if os.path.getmtime(loc) < time.time() - 24 * 60 * 60:
-            # File exists but older than one day — update it
-            _download_domain_list(loc)
-    else:
-        _download_domain_list(loc)
+    loc = os.path.join(bot.settings.homedir, "unsafedomains.txt")
+    if not os.path.isfile(loc) or os.path.getmtime(loc) < time.time() - 24 * 60 * 60:
+        # File doesn't exist or is older than one day — update it
+        url = bot.settings.safety.domain_blocklist_url
+        if url is None or not url.startswith("http"):
+            url = "https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts"
+        LOGGER.info("Downloading malicious domain list from %s", url)
+        # TODO: Can we use a cache header to avoid the download if it's unmodified?
```

Comment:
Is the possibility of doing so really in question, or is this actually a TODO asking some future person to implement checking said header?

Checking the output of `curl --head $stevenblack_url` shows that GitHub serves an [ETag header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag), which can be saved and compared before making future requests for the full file body.

```suggestion
        # TODO: Use ETag header to avoid the download if it's unmodified
```

Reply:
Mostly a suggestion for a future improvement, since it was beyond the scope of my original unfscking goals. I guess that could be easily stuck in the DB.